auth/test_jwt.py

```diff
@@ -0,0 +1,191 @@
+                            ("eyJBbGciOjIsIlRva2VuVHlwIjoxfQ", pytest.raises(ValueError)),
+                            ("", pytest.raises(ValueError)),
+                         ]
+                         )
+def test_validate_token_len(input_token, expectation):
+    with expectation:
+        jwt.validate_token_len(input_token)
+
+
+@pytest.mark.parametrize("input_str, expect_output, expectation",
+                         [
+                            ("", {}, pytest.raises(ValueError)),
+                            ("eyJBbGciOjIsIlRva2VuVHlwIjoxfQ", {'Alg': 2, 'TokenTyp': 1}, does_not_raise()),
+                            ("eyJVVUlEIjoiMDFkM3gzd20ybm5yZGZ6cDB0a2Eydnc5ZHgiLCJQZXJtaXNzaW9uIjozLCJFeHBpcmF0aW9uVGltZXN0YW1wIjoxODkzNDU2MDAwfQ", {'UUID': '01d3x3wm2nnrdfzp0tka2vw9dx', 'Permission': 3, 'ExpirationTimestamp': 1893456000}, does_not_raise()),
+                         ]
+                         )
+def test_base64_decode(input_str, expect_output, expectation):
+    with expectation:
+        actual_output = jwt.base64_decode(input_str)
+        assert actual_output == expect_output
+
+
+@pytest.mark.parametrize("input_str, expect_output",
+                         [
+                             ("eyJBbGciOjIsIlRva2VuVHlwIjoxfQ", {'Alg': 2, 'TokenTyp': 1}),
+                         ]
+                         )
+def test_get_decoded_header(input_str, expect_output):
+    actual_output = jwt.get_decoded_header(input_str)
+    assert actual_output == expect_output
+
+
+@pytest.mark.parametrize("input_str, expect_output",
+                         [
+                            ("eyJVVUlEIjoiMDFkM3gzd20ybm5yZGZ6cDB0a2Eydnc5ZHgiLCJQZXJtaXNzaW9uIjozLCJFeHBpcmF0aW9uVGltZXN0YW1wIjoxODkzNDU2MDAwfQ", {'UUID': '01d3x3wm2nnrdfzp0tka2vw9dx', 'Permission': 3, 'ExpirationTimestamp': 1893456000}),
+                         ]
+                         )
+def test_get_decoed_body(input_str, expect_output):
+    actual_output = jwt.get_decoded_body(input_str)
+    assert actual_output == expect_output
+
+
+@pytest.mark.parametrize("input_header_dict, expectation",
+                         [
+                             ("", pytest.raises(ValueError)),
+                             ({'Alg': -1, 'TokenTyp': 1}, pytest.raises(ValueError)),
+                             ({'Alg': 3, 'TokenTyp': 1}, pytest.raises(ValueError)),
+                             ({'Alg': 1, 'TokenTyp': -1}, pytest.raises(ValueError)),
+                             ({'Alg': 1, 'TokenTyp': 3}, pytest.raises(ValueError)),
+                             ({'Alg': 0, 'TokenTyp': 1}, does_not_raise()),
+                             ({'Alg': 1, 'TokenTyp': 1}, does_not_raise()),
+                             ({'Alg': 2, 'TokenTyp': 1}, does_not_raise()),
+                             ({'Alg': 1, 'TokenTyp': 0}, does_not_raise()),
+                             ({'Alg': 1, 'TokenTyp': 1}, does_not_raise()),
+                             ({'Alg': 1, 'TokenTyp': 2}, does_not_raise()),

```

Comment:
For all these ints, we should use enums. Hardcoding the int will break the tests if a new Alg or TokenTyp has been added